CSVReader/CSVReaders.rb

```diff
@@ -0,0 +1,42 @@
+class CSVReader
+  require "csv"
+
+  def self.inherited(subclass)
+    @subclass = subclass
+    @csv_container = CSV.read("#{@subclass}.csv", :headers => true).to_a
+    @headers = CSV.open("#{@subclass}.csv", 'r') { |csv| csv.first }
+    @headers.each do |method| 
+      class_eval("attr_accessor :#{method}") 
+
+      class_eval %Q"
+        def #{@subclass}.find_by_#{method}(argument)
+          #{@csv_container}.each do |row|
+            0.upto((row.length)-1) do |index|  
+              if(row[index] == argument)
+                process_results(row)
+              end
+            end
+          end
+        end
+      "
+
+      class_eval %Q"
+        def #{@subclass}.process_results(row)
+          subclass_object = #{@subclass}.new 
+          p subclass_object
+          p subclass_object = row
+        end
```

Comment:
Also when you do a find_by_xxx  it does return an object. But are you sure that  attributes (columns names based accessors) have those values.. 

emp = Employee.find_by_department("Finance") 
p emp.name
p emp.age
p emp.department

These attributes should return you the matched result